design_documents/test_design.md

```diff
@@ -0,0 +1,30 @@
+# Test design
+
+## Context
+
+Current test design in Starknet Forge has some issues:
+- gas estimation is problematic since we have to distinguish between pure Cairo gas
+(e.g., creating an array in test) and Starknet gas
+- starknet functions like `get_caller_address` are unavailable in tests directly
+- starknet system calls like `library_call` are unavailable in tests directly
+- testing internal contract functions requires exposing them as `external`
+- operating on the contract execution layer while tests being pure Cairo code can create confusion
```

Comment:
You mean with the current approach? We aren't really operating at this layer if I'm understanding correctly

Reply:
We are currently operating on execution layer (layer 2) with weird mix of pure Cairo (layer 1), am I right @MaksymilianDemitraszek?

Reply:
We are operating on layer 2